File: gitlab-nic/code-samples-main/API/Python/request-authcode.py
#!/usr/bin/env python3

# Requests an authorization code. 

import requests
import json
import sys

import afnic

client = "CTC71772"

if len(sys.argv) != 2:
    raise Exception("Usage: %s domain-name" % sys.argv[0])
domain = sys.argv[1]
headers = afnic.get_headers()
payload = {'domainName': domain, 'registrantClientId': client, 'justification': "Parce que c'est mon PROOOOJET !"}
response = requests.post("https://api-sandbox.nic.fr/v1/registrar/authorization-code-requests",
                         headers=headers, data=json.dumps(payload))
body = json.loads(response.text)
if response.status_code != 201:
    raise Exception("HTTP status code for auth. code request is %s (reason \"%s\")" % (response.status_code, body))
print("Authorization code %s requested" % body["repositoryObjectId"])


    

File: gitlab-nic/code-samples-main/API/Elixir/request-authcode.exs
# Requests an authorization code. 

url = "https://api-sandbox.nic.fr/v1/registrar/authorization-code-requests"

if length(System.argv()) < 1 do
  raise "Boom Usage: program domain-name"
end

client = "CTC71772"
domain = String.trim(List.first(System.argv()))

args =
  Jason.encode!(%{
    domainName: domain,
    registrantClientId: client,
    justification: "Parce que c'est mon PROOOOJET !"
  })

result = HTTPoison.post!(url, args, Afnic.get_headers(), [])

case result do
  %HTTPoison.Response{status_code: 201, body: body} ->
    response = Jason.decode!(body)
    IO.puts("Code for #{domain} requested, operation ID is #{response["repositoryObjectId"]}")

  %HTTPoison.Response{status_code: 400, body: body} ->
    response = Jason.decode!(body)
    IO.puts("Code not requested because #{List.first(response["errors"])["message"]}")

  other ->
    raise "Boom (#{inspect(other)})"
end
